pkg/controller/controller.go

```diff
@@ -0,0 +1,200 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
```

Comment:
Can we just `log/slog` instead of logrus on something new like this?

Reply:
While using something from the standard library makes sense, I'm disappointed by the user experience of slog. If `slog.Logger` had been an interface of some sort, I would have agreed.

As it is, introducing `slog` in a library means that any application that wants to use the library has to make changes in large, unrelated, parts of the application to accommodate the `slog` use in the library. That's a bad call, which is going to cause problems in the Go eco-system down the line. You just have to look at the history of logging in the JVM eco-system to see this exact kind of issue, and the problems that results from it.

Unfortunately, the reverse isn't much better, because the `log.FieldLogger` interface is badly defined. Had it not been for a few return values, it would have been simple to implement `log.FieldLogger` using a `slog.Logger` instance. :disappointed: 

As it is, I favor logrus for now, because that is what most (all?) of our operators so far is using.